wallet-test/src/test/scala/org/bitcoins/wallet/dlc/DLCNegotiationTest.scala

```diff
@@ -0,0 +1,101 @@
+
+import org.bitcoins.dlc.DLCMessage
+import org.bitcoins.testkit.wallet.FundWalletUtil.FundedWallet
+import org.bitcoins.testkit.wallet.{BitcoinSWalletTest, DLCWalletUtil}
+import org.scalatest.FutureOutcome
+
+class DLCNegotiationTest extends BitcoinSWalletTest {
+  type FixtureParam = FundedWallet
+
+  override def withFixture(test: OneArgAsyncTest): FutureOutcome = {
+    withFundedSegwitWallet(test)
+  }
+
+  behavior of "DLCNegotiation"
+
+  it must "create a dlc offer" in { funded: FundedWallet =>
+    val fundedWallet = funded.wallet
+
+    val offerData = DLCWalletUtil.sampleDLCOffer
+
+    for {
+      offer <- fundedWallet.createDLCOffer(
+        offerData.oracleInfo,
+        offerData.contractInfo,
+        offerData.totalCollateral,
+        Some(offerData.feeRate),
+        offerData.timeouts.contractMaturity.toUInt32,
+        offerData.timeouts.contractTimeout.toUInt32
+      )
+    } yield {
+      assert(offer.oracleInfo == offerData.oracleInfo)
+      assert(offer.contractInfo == offerData.contractInfo)
+      assert(offer.totalCollateral == offerData.totalCollateral)
+      assert(offer.feeRate == offerData.feeRate)
+      assert(offer.timeouts == offerData.timeouts)
+      assert(offer.fundingInputs.nonEmpty)
+      assert(offer.changeAddress.value.nonEmpty)
+    }
+  }
+
+  it must "accept a dlc offer" in { funded: FundedWallet =>
+    val fundedWallet = funded.wallet
+
+    val offer = DLCWalletUtil.sampleDLCOffer
+    val eventId = DLCMessage.calcEventId(offer.oracleInfo,
+                                         offer.contractInfo,
+                                         offer.timeouts)
+
+    for {
+      accept <- fundedWallet.acceptDLCOffer(offer)
+    } yield {
+      assert(accept.fundingInputs.nonEmpty)
+      assert(accept.eventId == eventId)
+      assert(
+        accept.totalCollateral == offer.contractInfo.values.max - offer.totalCollateral)
+      assert(accept.changeAddress.value.nonEmpty)
+    }
+  }
+
+  it must "sign a dlc" in { funded: FundedWallet =>
+    val fundedWallet = funded.wallet
+
+    val offerData = DLCWalletUtil.sampleDLCOffer
+    val accept = DLCWalletUtil.sampleDLCAccept
+
+    for {
+      _ <- fundedWallet.createDLCOffer(
+        offerData.oracleInfo,
+        offerData.contractInfo,
+        offerData.totalCollateral,
+        Some(offerData.feeRate),
+        offerData.timeouts.contractMaturity.toUInt32,
+        offerData.timeouts.contractTimeout.toUInt32
+      ) // must initialize the dlc in the database
+
+      sign <- fundedWallet.signDLC(accept)
+    } yield {
+      assert(sign.eventId == accept.eventId)
+      assert(sign.fundingSigs.nonEmpty)
+    }
+  }
+
+  it must "add dlc sigs" in { funded: FundedWallet =>
+    val fundedWallet = funded.wallet
+
+    val offer = DLCWalletUtil.sampleDLCOffer
+    val sigs = DLCWalletUtil.sampleDLCSign
+
+    for {
+      _ <- fundedWallet.acceptDLCOffer(offer) // must initialize the dlc in the database
```

Comment:
We might want to consider making `initDLC` package private rather than fully private to avoid this hack? I guess I don't have strong feelings about it but it seems like the cleaner option

Reply:
Doesn't really solve the problem because we need the funding inputs in the database or some of the offer data